src/NHibernate/Type/AbstractStringType.cs

```diff
@@ -1,13 +1,42 @@
 using System;
+using System.Collections.Generic;
 using System.Data.Common;
+using System.Globalization;
 using NHibernate.Engine;
 using NHibernate.SqlTypes;
+using NHibernate.UserTypes;
 
 namespace NHibernate.Type
 {
 	[Serializable]
-	public abstract class AbstractStringType: ImmutableType, IDiscriminatorType
+	public abstract class AbstractStringType: ImmutableType, IDiscriminatorType, IParameterizedType
 	{
+		/// <summary>
+		/// The comparer culture parameter name. Value should be <c>Current</c>, <c>Invariant</c>,
+		/// <c>Ordinal</c> or any valid culture name.
+		/// </summary>
+		/// <remarks>Default comparison is ordinal.</remarks>
+		public const string ComparerCultureParameterName = "ComparerCulture";
+
+		/// <summary>
+		/// The case sensitivity parameter name. Value should be a boolean, <c>false</c> meaning
+		/// case insensitive.
+		/// </summary>
+		/// <remarks>Default comparison is case sensitive.</remarks>
+		public const string CaseSensitiveParameterName = "CaseSensitive";
+
+		/// <summary>
+		/// The default string comparer for string equality and hashcodes. <see langword="null" /> for
+		/// using .Net default equality and hashcode computation.
+		/// </summary>
+		public static StringComparer DefaultComparer { get; set; }
+
+		/// <summary>
+		/// The string comparer of this instance of string type, for string equality and hashcodes.
+		/// <see langword="null" /> for using <see cref="DefaultComparer"/>.
+		/// </summary>
+		protected StringComparer Comparer { get; set; }
```

Comment:
I have considered exposing this publicly, for allowing user to set the `Comparer` by code on a specific type instance. But since the type factory get type methods may yield cached instances, it would be a trap.